packages/react/src/components/ThemeProvider/index.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+
+import { createTheme, Theme } from '@aws-amplify/ui';
+
+import { ThemeContext, ColorMode } from './ThemeContext';
+
+interface ThemeProviderProps {
+  children: React.ReactNode;
+  colorMode?: ColorMode;
+  theme?: Theme;
+  root?: boolean;
```

Comment:
I'm slightly concerned with this prop leading to user confusion, but i may not fully grasp the need for it?